packages/statemanager/test/rpcStateManager.spec.ts

```diff
@@ -0,0 +1,339 @@
+    const vm = await VM.create({ common, stateManager: <any>state }) // TODO fix the type DefaultStateManager back to StateManagerInterface in VM
+
+    const vitalikDotEth = Address.fromString('0xd8da6bf26964af9d7eed9e03e53415d37aa96045')
+    const privateKey = hexToBytes(
+      '0xe331b6d69882b4cb4ea581d88e0b604039a3de5967688d3dcffdd2270c0fd109'
+    )
+    const tx = FeeMarketEIP1559Transaction.fromTxData(
+      { to: vitalikDotEth, value: '0x100', gasLimit: 500000n, maxFeePerGas: 7 },
+      { common }
+    ).sign(privateKey)
+
+    const result = await vm.runTx({
+      skipBalance: true,
+      skipNonce: true,
+      tx,
+    })
+
+    assert.equal(result.totalGasSpent, 21000n, 'sent some ETH to vitalik.eth')
+  })
+})
+
+describe('runTx test: replay mainnet transactions', () => {
+  it('should work', async () => {
+    const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
+
+    const blockTag = 15496077n
+    common.setHardforkBy({ blockNumber: blockTag })
+    const tx = await TransactionFactory.fromRPC(txData, { common })
+    const state = new RPCStateManager({
+      provider,
+      // Set the state manager to look at the state of the chain before the block has been executed
+      blockTag: blockTag - 1n,
+    })
+    const vm = await VM.create({ common, stateManager: <any>state })
+    const res = await vm.runTx({ tx })
+    assert.equal(
+      res.totalGasSpent,
+      21000n,
+      'calculated correct total gas spent for simple transfer'
+    )
+  })
+})
+
+describe('runBlock test', () => {
+  it('should work', async () => {
+    const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Chainstart })
+
+    const blockTag = 500000n
+    const state = new RPCStateManager({
+      provider,
+      // Set the state manager to look at the state of the chain before the block has been executed
+      blockTag: blockTag - 1n,
+    })
+
+    // Set the common to HF, doesn't impact this specific blockTag, but will impact much recent
+    // blocks, also for post merge network, ttd should also be passed
+    common.setHardforkBy({ blockNumber: blockTag - 1n })
+
+    const vm = await VM.create({ common, stateManager: state })
+    const block = Block.fromRPC(blockData, [], { common })
+    try {
+      const res = await vm.runBlock({
+        block,
+        generate: true,
+        skipHeaderValidation: true,
+      })
+      assert.equal(
+        res.gasUsed,
+        block.header.gasUsed,
+        'should compute correct cumulative gas for block'
+      )
+    } catch (err: any) {
+      assert.fail(`should have successfully ran block; got error ${err.message}`)
+    }
+  })
+})
+
+describe('blockchain', () =>
+  it('uses blockhash', async () => {
+    const blockchain = new RPCBlockChain(provider)
+    const blockTag = 1n
+    const state = new RPCStateManager({ provider, blockTag })
+    const evm = new EVM({ blockchain, stateManager: state })
+    const code = '0x600143034060005260206000F3'
```

Comment:
Nit: small comment what this does (return blockhash of previous block - right?)

Reply:
Agreed.  Will add.